static/js/reducers/email.js

```diff
@@ -15,38 +16,57 @@ import {
 } from '../actions';
 import type { Action } from '../flow/reduxTypes';
 import type {
-  Email,
-  EmailState
+  AllEmailsState,
+  EmailState,
+  EmailInputs,
 } from '../flow/emailTypes';
 
 export const INITIAL_EMAIL_STATE: EmailState = {
-  email:  {},
+  inputs: {},
   validationErrors: {},
   sendError: {},
 };
+export const INITIAL_ALL_EMAILS_STATE: AllEmailsState = {
+  searchResultEmail: INITIAL_EMAIL_STATE,
+  courseTeamEmail: INITIAL_EMAIL_STATE
+};
 
-export const NEW_EMAIL_EDIT: Email = {
+export const NEW_EMAIL_EDIT: EmailInputs = {
   subject:    null,
   body:       null,
-  query:      null,
 };
 
-export const email = (state: EmailState = INITIAL_EMAIL_STATE, action: Action) => {
+function updatedState(state: AllEmailsState, emailType: string, updateObject: Object) {
+  _.merge(state[emailType], updateObject);
+  return _.cloneDeep(state);
+}
+
+export const email = (state: AllEmailsState = INITIAL_ALL_EMAILS_STATE, action: Action) => {
+  let emailType;
+
   switch (action.type) {
   case START_EMAIL_EDIT:
-    return { ...state, email: { ...NEW_EMAIL_EDIT, query: action.payload } };
+    emailType = action.payload;
+    return updatedState(state, emailType, { inputs: NEW_EMAIL_EDIT });
   case UPDATE_EMAIL_EDIT:
-    return { ...state, email: action.payload };
+    emailType = action.payload.type;
+    return updatedState(state, emailType, { inputs: action.payload.inputs });
```

Comment:
that works! my intent with setting the variable was just to make it clear to someone reading the code that the type could be found in different parts of the payload depending on the action. that helper function does the job a little better